### DISCUSSION TITLE
The "dot" operator seems to be missing from the spec.

### DISCUSSION THREAD

**Discussion by seanmcl** [openxla/community#119] [stablehlo]
We're trying to implement a spec/parser/etc for [StableHLO in Lean](https://github.com/leanprover/SHerLOC). Reading https://openxla.org/stablehlo/spec#ops and the corresponding github file, I don't see docs for `dot`. There is `dot_generic`. Should the later somehow subsume the former, or should `dot` be in the spec? If the former, there are some examples with `dot` in the stablehlo repo that should be updated.

Thanks!

**Comment by ghpvnist** [comment]
StableHLO historically inherited `dot` from MHLO, and since `dot_general` can fully express `dot`, we plan to deprecate the op from StableHLO. See https://github.com/openxla/stablehlo/issues/3 for original issue, RFC for its [future](https://github.com/openxla/stablehlo/pull/2283), and current documentation [here](https://github.com/openxla/stablehlo/blob/15324c4d313556d0d1324a33ce66e9172534fd1d/stablehlo/dialect/StablehloOps.td#L2425-L2429).

Edit: We also have a MHLO [pass](https://github.com/openxla/xla/blob/main/xla/mlir_hlo/mhlo/transforms/legalize_dot_to_dot_general/legalize_dot_to_dot_general.cc) to convert `dot` to `dot_general`.

**Comment by seanmcl** [comment]
I see. Thanks!

I got the `dot` from the [first example in the spec](https://github.com/openxla/stablehlo/blob/main/docs/spec.md). I guess it will just take a while to expunge all these operators from the different docs and tools.

```
```mlir
func.func @main(
  %image: tensor<28x28xf32>,
  %weights: tensor<784x10xf32>,
  %bias: tensor<1x10xf32>
) -> tensor<1x10xf32> {
  %0 = "stablehlo.reshape"(%image) : (tensor<28x28xf32>) -> tensor<1x784xf32>
  %1 = "stablehlo.dot"(%0, %weights) : (tensor<1x784xf32>, tensor<784x10xf32>) -> tensor<1x10xf32>
  %2 = "stablehlo.add"(%1, %bias) : (tensor<1x10xf32>, tensor<1x10xf32>) -> tensor<1x10xf32>
  %3 = "stablehlo.constant"() {value = dense<0.0> : tensor<1x10xf32>} : () -> tensor<1x10xf32>
  %4 = "stablehlo.maximum"(%2, %3) : (tensor<1x10xf32>, tensor<1x10xf32>) -> tensor<1x10xf32>
  "func.return"(%4): (tensor<1x10xf32>) -> ()
}
```
